server/routes/referrals/referralFormPresenter.ts

```diff
@@ -4,120 +4,125 @@ export default class ReferralFormPresenter {
-            title: 'Enter RAR days used',
-            url: 'rar-days',
-          },
-          {
-            title: 'Further information for service provider',
-            url: 'further-information',
-          },
-        ],
-      },
-      {
-        type: 'single',
-        title: 'Review responsible officer’s information',
-        number: '3',
-        status: responsibleOfficerDetailsSectionFormStatus,
-        tasks: [
-          {
-            title: 'Responsible officer information',
-            url: null,
-          },
-        ],
-      },
-      {
-        type: 'single',
-        title: 'Check your answers',
-        number: '4',
-        status: checkAnswersSectionFormStatus,
-        tasks: [
-          {
-            title: 'Check your answers',
-            url: checkAnswersSectionFormStatus !== ReferralFormStatus.CannotStartYet ? 'check-answers' : null,
-          },
-        ],
-      },
-    ]
+  private calculateTaskUrl(url: string | null, displayCriteria: DraftReferralValues): string | null {
+    const everyFieldHasValues = displayCriteria.every(field => {
+      if (field === null) {
+        return false
+      }
+      return Array.isArray(field) ? field.length !== 0 : true
+    })
+    return everyFieldHasValues ? url : null
   }
 
-  private referralFormStatus(
+  private calculateStatus(
     draftReferralValues: DraftReferralValues,
-    previousFormStatus: ReferralFormStatus = ReferralFormStatus.Completed
+    previousSectionStatus: ReferralFormStatus = ReferralFormStatus.Completed
   ): ReferralFormStatus {
-    if (previousFormStatus !== ReferralFormStatus.Completed) {
+    if (previousSectionStatus !== ReferralFormStatus.Completed) {
       return ReferralFormStatus.CannotStartYet
     }
-    const hasCompleted = draftReferralValues.every(field => {
+    const allFieldsHaveValues = draftReferralValues.every(field => {
       if (field === null) {
         return false
       }
       return Array.isArray(field) ? field.length !== 0 : true
     })
-    if (hasCompleted) {
+    if (allFieldsHaveValues) {
       return ReferralFormStatus.Completed
     }
     return ReferralFormStatus.NotStarted
   }
 
-  private get requiredServiceUserFormValues(): DraftReferralValues {
+  private get reviewServiceUserInformationSectionValues(): DraftReferralValues {
+    return this.serviceUserDetailsTaskValues && this.riskInformationTaskValues && this.needsAndRequirementsTaskValues
+  }
+
+  private get serviceCategoryReferralDetailsSectionValues(): DraftReferralValues {
+    return (
+      this.relevantSentenceTaskValues &&
+      this.desiredOutcomesTaskValues &&
+      this.complexityLevelTaskValues &&
+      this.completionDeadlineTaskValues &&
+      this.rarDaysTaskValues &&
+      this.furtherInformationTaskValues
+    )
+  }
+
+  private get checkYourAnswersSectionValues(): DraftReferralValues {
+    return this.checkAnswersTaskValues
+  }
+
+  private get serviceUserDetailsTaskValues(): DraftReferralValues {
```

Comment:
I was testing this out locally with CRN `X320741` (one that has a sentence available to select from the local community API) - I couldn't figure out why the second link wouldn't become visible, and it's because the test data doesn't have a value for all of these fields - see screenshot:

![image](https://user-images.githubusercontent.com/19826940/117304983-3d22d900-ae76-11eb-87c2-7477f09d7323.png)

I don't think this is a very realistic example of data we'll get from the production APIs, but it suggests that those values _could_ be `null`, in which case the SU details tasks will never be completed. I wonder if it's worth just putting a check on CRN, FirstName, date of birth? It's hard to know which fields will always be present in the response though...

I suppose if we were to pick up this ticket: https://dsdmoj.atlassian.net/browse/IC-1245, we could set the values to `None` or `N/A` or something like that, but I'm not sure the priority of that ticket. What do you think?

Reply:
Maybe, like you demoed this morning, we could just always make that link available as a first step?